qiskit/visualization/dag_visualization.py

```diff
@@ -215,16 +264,53 @@ def edge_attr_func(edge):
             e["label"] = label
             return e
 
-    image_type = None
+    image_type = "png"
     if filename:
         if "." not in filename:
             raise InvalidFileError("Parameter 'filename' must be in format 'name.extension'")
         image_type = filename.split(".")[-1]
-    return graphviz_draw(
-        dag._multi_graph,
-        node_attr_func,
-        edge_attr_func,
-        graph_attrs,
-        filename,
-        image_type,
-    )
+        if image_type not in IMAGE_TYPES:
+            raise ValueError(
+                "The specified value for the image_type argument, "
+                f"'{image_type}' is not a valid choice. It must be one of: "
+                f"{IMAGE_TYPES}"
+            )
+
+    if isinstance(dag, DAGCircuit):
+        dot_str = dag._to_dot(
+            graph_attrs,
+            node_attr_func,
+            edge_attr_func,
+        )
+
+        prog = "dot"
+        if not filename:
+            dot_result = subprocess.run(
+                [prog, "-T", image_type],
+                input=dot_str.encode("utf-8"),
+                capture_output=True,
+                encoding=None,
+                check=True,
+                text=False,
+            )
+            dot_bytes_image = io.BytesIO(dot_result.stdout)
+            image = Image.open(dot_bytes_image)
+            return image
+        else:
+            subprocess.run(
+                [prog, "-T", image_type, "-o", filename],
+                input=dot_str,
+                check=True,
+                encoding="utf8",
+                text=True,
+            )
+            return None
+    else:
+        return graphviz_draw(
+            dag._multi_graph,
+            node_attr_func,
+            edge_attr_func,
+            graph_attrs,
+            filename,
+            image_type,
+        )
```

Comment:
The dag visualization is used for dag dependency too, and that still relies on the rustworkx `graphviz_draw` to work (which was a surprise to me when I first did this, I probably just forgot). The `DAGCircuit` path needs to be handled on it's own because it owns generating it's own dot representation now, while dag dependency still relies on rustworkx.

Reply:
In that case, the function's argument documentation is slightly out of date (probably long-standing, but it tripped me up here).

Reply:
Tracked in #13021.